src/solveh.rs

```diff
@@ -346,50 +340,129 @@ where
     (sign, ln_det)
 }
 
-impl<A, S> DeterminantH for BKFactorized<S>
+impl<A, S> BKFactorized<S>
 where
     A: Scalar,
     S: Data<Elem = A>,
 {
-    type Output = A::Real;
-    type SignLnOutput = (A::Real, A::Real);
-
-    fn deth(&self) -> A::Real {
+    /// Computes the determinant of the factorized Hermitian (or real
+    /// symmetric) matrix.
+    pub fn deth(&self) -> A::Real {
         let (sign, ln_det) = self.sln_deth();
         sign * ln_det.exp()
     }
 
-    fn sln_deth(&self) -> (A::Real, A::Real) {
+    /// Computes the `(sign, natural_log)` of the determinant of the factorized
+    /// Hermitian (or real symmetric) matrix.
+    ///
+    /// The `natural_log` is the natural logarithm of the absolute value of the
+    /// determinant. If the determinant is zero, `sign` is 0 and `natural_log`
+    /// is negative infinity.
+    ///
+    /// To obtain the determinant, you can compute `sign * natural_log.exp()`
+    /// or just call `.deth()` instead.
+    ///
+    /// This method is more robust than `.deth()` to very small or very large
+    /// determinants since it returns the natural logarithm of the determinant
+    /// rather than the determinant itself.
+    pub fn sln_deth(&self) -> (A::Real, A::Real) {
         bk_sln_det(UPLO::Upper, self.ipiv.iter().cloned(), &self.a)
     }
 }
 
-impl<A, S> DeterminantHInto for BKFactorized<S>
+impl<A, S> DeterminantH<A> for BKFactorized<S>
```

Comment:
Hum... This implementation is redundant.
We should replace `impl DeterminantH{Into} for BKFactorized` by the associated function `BKFactorized::deth(&self)` as you did.

Reply:
The reason why I kept the `DeterminantH*` implementations for `BKFactorized` is in case someone wanted a function that's generic with a `DeterminantH` bound to work with either `ArrayBase` or `BKFactorized`, e.g.:

```rust
fn determinant_squared<D>(something: &D) -> f64
where
    D: DeterminantH<Elem = f64>,
{
    something.deth().unwrap().powi(2)
}
```

However, this is probably a rare case, and I can certainly remove the `DeterminantH*` implementations for `BKFactorized` if that's what you'd prefer.